pages/my-plants.js

```diff
@@ -1,8 +1,10 @@
 import PlantCard from "@/components/PlantCard";
 import { PlantList } from ".";
+import useSWR from "swr";
 
-export default function MyPlants({ ownedPlantsIds, plants, onToggleOwned }) {
-  const myPlants = plants.filter((plant) => ownedPlantsIds.includes(plant.id));
+export default function MyPlants({ ownedPlantsIds, onToggleOwned }) {
+  const { data: plants } = useSWR("/api/plants");
+  const myPlants = plants.filter((plant) => ownedPlantsIds.includes(plant._id));
 
```

Comment:
You should implement error handling here as well (error and isLoading), because someone could directly give this url to the browser.